tests/int/test_headers.py

```diff
@@ -0,0 +1,174 @@
+    "BUSINESS_FUNCTION" : "nhsd-ers-business-function",
+    "ODS_CODE" : "nhsd-ers-ods-code",
+    "FILENAME" : "nhsd-ers-file-name",
+    "COMM_RULE_ORG" : "nhsd-ers-comm-rule-org",
+    "REFERRAL_ID" : "nhsd-ers-referral-id"
+}
+
+RENAMED_HEADERS = {
+    "CORRELATION_ID" : "nhsd-correlation-id",
+    "BUSINESS_FUNCTION" : "x-ers-business-function",
+    "ODS_CODE" : "x-ers-ods-code",
+    "FILENAME" : "x-ers-xapi-meta-file_name",
+    "COMM_RULE_ORG" : "x-ers-xapi-comm-rule-org",
+    "REFERRAL_ID" : "x-ers-xapi-meta-intended_ubrn"
+}
+
+@pytest.mark.smoke_test
+class TestHeaders:
+
+    @pytest.fixture
+    async def product(self):
+        # Setup
+        product = ApigeeApiProducts()
+        await product.create_new_product()
+
+        print('GOT PRODUCT NAME: '  + product.name)
+
+        # Update products allowed paths
+        await product.update_proxies([SERVICE_NAME, 'identity-service-' + ENVIRONMENT])
+
+        await product.update_scopes(
+            scopes=["urn:nhsd:apim:user-nhs-id:aal3:e-referrals-service-api"]
+        )
+        yield product
+
+        #Teardown
+        print('Cleanup product')
+        await product.destroy_product()
+
+
+    @pytest.fixture
+    async def app(self, asid = '1234567890'):
+         # Setup
+        app = ApigeeApiDeveloperApps()
+        await app.create_new_app()
+
+        # Set default JWT Testing resource url and ASID
+        await app.set_custom_attributes({
+                'asid': asid,
+                'jwks-resource-url': 'https://raw.githubusercontent.com/NHSDigital/'
+                                     f'identity-service-jwks/main/jwks/{ENVIRONMENT}/'
+                                     '9baed6f4-1361-4a8e-8531-1f8426e3aba8.json'
+        })
+        yield app
+
+        #Teardown
+        print('Cleanup app')
+        await app.destroy_app()
+
+
+    async def get_access_code(self, appToUse, user_id):
+        # Create an OAuthHelper instance with the new app
+        oauth = OauthHelper(appToUse.client_id, appToUse.client_secret, appToUse.callback_url)
+
+        with open(ID_TOKEN_TESTING_PRIVATE_KEY_PATH, "r") as f:
+            contents = f.read()
+
+        claims = {
+                'at_hash': 'tf_-lqpq36lwO7WmSBIJ6Q',
+                'sub': user_id,
+                'auditTrackingId': '91f694e6-3749-42fd-90b0-c3134b0d98f6-1546391',
+                'amr': ['N3_SMARTCARD'],
+                'iss': 'https://am.nhsint.auth-ptl.cis2.spineservices.nhs.uk:443/'
+                       'openam/oauth2/realms/root/realms/NHSIdentity/realms/Healthcare',
+                'tokenName': 'id_token',
+                'aud': '969567331415.apps.national',
+                'c_hash': 'bc7zzGkClC3MEiFQ3YhPKg',
+                'acr': 'AAL3_ANY',
+                'org.forgerock.openidconnect.ops': '-I45NjmMDdMa-aNF2sr9hC7qEGQ',
+                's_hash': 'LPJNul-wow4m6Dsqxbning',
+                'azp': '969567331415.apps.national',
+                'auth_time': 1610559802,
```

Comment:
Would be good to have this programmatic, as you do for exp and iat.

Reply:
I actually have no idea what auth_time means but searching through the other examples they have the same sort of number.

Reply:
This is the time the authentication occurred. I have added a new review comment for this so it doesn't get lost in this thread.